lib/index.ts

```diff
@@ -293,40 +352,9 @@ export default class Backburner {
     return this.later(...arguments);
   }
 
-  public later(...args) {
-    let length = args.length;
-
-    let wait = 0;
-    let method;
-    let target;
-    let methodOrTarget;
-    let methodOrArgs;
-
-    if (length === 0) {
-      return;
-    } else if (length === 1) {
-      method = args.shift();
-    } else {
-      let last = args[args.length - 1];
-
-      if (isCoercableNumber(last)) {
-        wait = parseInt(args.pop(), 10);
-      }
-
-      methodOrTarget = args[0];
-      methodOrArgs = args[1];
-      let type = typeof methodOrArgs;
-      if (type === 'function') {
-        target = args.shift();
-        method = args.shift();
-      } else if (methodOrTarget !== null && type === 'string' && methodOrArgs in methodOrTarget) {
-        target = args.shift();
-        method = target[args.shift()];
-      } else {
-        method = args.shift();
-      }
-    }
-
+  public later(...args);
+  public later() {
+    let [target, method, args, wait] = parseTimerArgs(...arguments);
```

Comment:
is the transpiled `loose` for this statement or does this try to do the funky iterables capable output?

Reply:
If its loose, we are good. If not, we should make it loose.

Reply:
its all good I think, here is output https://cl.ly/07260l1y0a2k/Screen%20Shot%202018-01-24%20at%2012.28.50%20PM.png

Reply:
Ya, we should confirm how Ember consumes (it uses ES output so does its own transpile) also but I am pretty sure we use loose there too...

Reply:
yep same in ember https://cl.ly/3E0m1k3f463s/Screen%20Shot%202018-01-24%20at%202.19.44%20PM.png